langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/LangLibArrayTest.java

```diff
@@ -120,19 +120,39 @@ public void testSlice() {
         assertEquals(arr.getFloat(0), 23.45);
         assertEquals(arr.getFloat(1), 34.56);
         assertEquals(arr.getFloat(2), 45.67);
+        assertEquals(((BInteger) result.getRefValue(1)).intValue(), 3);
 
-        arr = (BValueArray) result.getRefValue(1);
+        arr = (BValueArray) result.getRefValue(2);
         assertEquals(arr.elementType.getTag(), TypeTags.FLOAT_TAG);
         assertEquals(arr.size(), 3);
         assertEquals(arr.getFloat(0), 34.56);
         assertEquals(arr.getFloat(1), 45.67);
         assertEquals(arr.getFloat(2), 56.78);
+        assertEquals(((BInteger) result.getRefValue(3)).intValue(), 3);
 
-        arr = (BValueArray) result.getRefValue(2);
+        arr = (BValueArray) result.getRefValue(4);
         assertEquals(arr.elementType.getTag(), TypeTags.FLOAT_TAG);
         assertEquals(arr.size(), 2);
         assertEquals(arr.getFloat(0), 45.67);
         assertEquals(arr.getFloat(1), 56.78);
+        assertEquals(((BInteger) result.getRefValue(5)).intValue(), 2);
+    }
+
+    @Test
+    public void testPushAfterSlice() {
```

Comment:
Can you add test cases for fixed-length arrays as well (if they aren't there)? Both positive and negative cases.

Reply:
Added test cases for fixed-length. 
Currently (with this PR) the result of slice function on a  fixed-length arrays also returns a non-fixed array. Hence, there aren't any negative cases. 

Once this https://github.com/ballerina-platform/ballerina-spec/issues/391#issuecomment-574986884 is clarified, i'll make the necessary changes and add the negative test cases.